File: 0231-power-of-two/0231-power-of-two.py
class Solution:
    def solve(self, n: int, i: int) -> bool:
        if(pow(2,i)==n):
            return True
        if(pow(2,i)>n):
            return False
        return self.solve(n,i+1)
        
    def isPowerOfTwo(self, n: int) -> bool:  
        if(n<=0):
            return False
        return self.solve(n,0)

File: 0231-power-of-two/0231-power-of-two.java
class Solution {
    private boolean solve(int n, int i)
    {
        if(Math.pow(2,i)==n)
        {
            return true;
        }
        if(Math.pow(2,i)>n)
        {
            return false;
        }
        return solve(n,i+1);
    }

    public boolean isPowerOfTwo(int n) {
        if(n<=0)
        {
            return false;
        }
        return solve(n,0);
    }
}